src/substrait/textplan/README.md

```diff
@@ -0,0 +1,208 @@
+# Substrait Text Plan Language Reference
+
+## Types
+Types are the core at the core of Substrait plans.  Functions consume data of a particular type and output data in the 
+same or a different type.  Optional types in Substrait are represented by appending a question mark to the type.
```

Comment:
```suggestion
Types are at the core of Substrait plans.  Functions consume data of a particular type and output data in the 
same or a different type.  Nullable types in Substrait are represented by appending a question mark to the type.
```

Small grammar change in the first sentence.  More importantly, can we replace "optional" with "nullable"?  I think that is more consistent with docs elsewhere.